### DISCUSSION TITLE
Use of global variables

### DISCUSSION THREAD

**Discussion by otsuka** [fastapi/fastapi#8155] [Questions] [answered]
I would like to know about the use of global variables within FastAPI application.

In aiohttp, the use of global variables is discouraged.
https://aiohttp.readthedocs.io/en/stable/web_advanced.html#data-sharing-aka-no-singletons-please

Does FastAPI have such a constraint?  Is it possible to use global variables?

For example, can the `database` global variable in this `main.py` be used safely by coroutines (`async def` functions) in the application?
https://fastapi.tiangolo.com/tutorial/async-sql-databases/#import-and-set-up-databases

**Comment by tiangolo** [comment, marked as answer]
Yep, you can use globals or anything you need.

**Comment by tiangolo** [comment]
BTW, `databases` is created by the same Tom Christie that created Starlette. So, Starlette (and that way FastAPI) is the main framework it is tested with.

**Comment by otsuka** [comment]
@tiangolo
Thank you for your answer!
